tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -116,30 +116,35 @@ def _extract_commands_from_example(example_text):
 
     # fold commands spanning multiple lines into one line. Split commands that use pipes
     example_text = example_text.replace("\\\n", " ")
-    example_text = example_text.replace("\\ ", " ")
-    example_text = example_text.replace(" | ", "\n")
+    example_text = example_text.replace("\\ ", " ")  # Handle cases where escpae is used to continue string on multiple line.
 
     commands = example_text.splitlines()
     processed_commands = []
     for command in commands:  # filter out commands
         if command.startswith("az"):
             processed_commands.append(command)
-        elif "az " in command:  # some commands start with "$(az ..." and even "`az in one case"
+        elif "az " in command:  # some commands start with "$(az ..." and even "`az in one case" deal with closing brackets later.
             idx = command.find("az ")
             command = command[idx:]
+            command = command.rstrip(")`")
```

Comment:
Let's prevent a scenario like:
```
az baz --foo $(az foo) --bar
```
from seeing `az foo) --bar` as a inner command.

We'll ignore more than one level of nestedness per our discussion :)